governance/program/src/state/proposal.rs

```diff
@@ -259,6 +259,22 @@ impl ProposalV2 {
         Ok(())
     }
 
+    /// Checks the Proposal was finalized (no more state transition will happen)
+    pub fn assert_is_finalized(&self) -> Result<(), ProgramError> {
```

Comment:
nit: `Finalised` can be confusing because `FinaliseVote` instruction is used to transition a `Proposal` in `Voting` state to either `Defeated` or `Succeeded`.
We could be more explicit about it and name the function `assert_is_final_state()` instead.